code/modules/credits/credits.dm

```diff
@@ -0,0 +1,184 @@
+#define CREDIT_ROLL_SPEED 100
+#define CREDIT_SPAWN_SPEED 20
+#define CREDIT_ANIMATE_HEIGHT (14 * world.icon_size)
+#define CREDIT_EASE_DURATION 22
+
+GLOBAL_VAR_INIT(end_credits_song, null)
+GLOBAL_VAR_INIT(end_credits_title, null)
+GLOBAL_LIST(end_titles)
+
+/datum/controller/subsystem/ticker/proc/generate_credits()
+
+	if(!GLOB.end_titles)
+		GLOB.end_titles = generate_titles()
+
+	for(var/client/C in GLOB.clients)
+		if(!C)
+			continue
+		if(!length(C.credits))
+			C.RollCredits()
+
+/client/proc/RollCredits()
+	if(!mob.get_preference(PREFTOGGLE_3_POSTCREDS))
+		return
+
+	verbs += /client/proc/ClearCredits
+	for(var/I in GLOB.end_titles)
+		if(!credits)
+			return
+		var/atom/movable/screen/credit/T = new(null, null, I, src)
+		credits += T
+		T.rollem()
+		sleep(CREDIT_SPAWN_SPEED)
+	sleep(CREDIT_ROLL_SPEED - CREDIT_SPAWN_SPEED)
```

Comment:
Why not put this sleep onto a timer? The one in the loop I can understand for the sake of timing, but I think one for just the credit replacement makes sense.